src/LinkDotNet.NCronJob/Configuration/SupportsConcurrencyAttribute.cs

```diff
@@ -1,10 +1,21 @@
 namespace LinkDotNet.NCronJob;
 
 /// <summary>
-/// Identifies the Job as allowing concurrent execution. This means that multiple Jobs of the same type can be run simultaneously.
+/// Specifies that multiple instances of the same job type can run simultaneously.
+/// This attribute controls the maximum degree of parallelism allowed for instances of the job.
 /// </summary>
 [AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
 public sealed class SupportsConcurrencyAttribute : Attribute
 {
+    /// <summary>
+    /// Gets the maximum number of concurrent instances allowed for the job.
+    /// </summary>
+    public int MaxDegreeOfParallelism { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SupportsConcurrencyAttribute"/> class with the specified maximum degree of parallelism.
+    /// </summary>
+    /// <param name="maxDegreeOfParallelism">The maximum number of concurrent instances that can run.</param>
+    public SupportsConcurrencyAttribute(int maxDegreeOfParallelism) => MaxDegreeOfParallelism = maxDegreeOfParallelism;
```

Comment:
I do get your intention and the code. 
Here are my thoughts, and I want your opinion on this one.

Currently, in your implementation, you check if `SupportsConcurrencyAttribute.MaxDegreeOfParallelism > GlobalConfig.MaxDegreeOfParallelism`, then we get an exception because it would give the theoretical possibility of having more jobs than slots available.

What if we move this into the scheduler/executor?
I am proposing this because now users have to provide a value. But imagine I have an idempotent job where I don't care how many instances run in parallel; still, I have to give an arbitrary value.
In this case I just want to flag the job, but don't care if there are 2, 3 or 5 instances in parallel.

So that would basically map to:
```csharp
[SupportsConcurrency(int.MaxValue)]
public class MyIdempotantJob : IJob
```

Basically the signature and class looks like this:
```csharp
public sealed class SupportsConcurrencyAttribute : Attribute
{
     public int MaxDegreeOfParallelism { get; set; } = int.MaxValue;
}
```

And people would resolve it like this:
```csharp
// I don't care how many instances run
[SupportConcurrency]
public class Job : IJob

// I only want 2 instances
[SupportConcurrency(MaxDegreeOfParallelism = 2]
public class Job : IJob
```

What is your take on this?

Reply:
> Currently, in your implementation, you check if `SupportsConcurrencyAttribute.MaxDegreeOfParallelism > GlobalConfig.MaxDegreeOfParallelism`, then we get an exception because it would give the theoretical possibility of having more jobs than slots available.
> 
> What if we move this into the scheduler/executor?

So I put this in the NCronJobOptionBuilder so that it throws immediately before anything gets a chance to be scheduled. However now thinking about it you may not want to keep the entire program from running just because one job is misconfigured. It's probably more reasonable to log a warning and automatically set the `concurrencyAttribute.MaxDegreeOfParallelism = settings.MaxDegreeOfParallelism` for that job.

I agree that some developers should be able to run a job without worrying about degrees of parallelism. We should implement a sensible default for such cases. Not sure if MaxValue is the way to go, maybe just keep it to the processor count and add a default ctor like this: 
```csharp
[AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
public sealed class SupportsConcurrencyAttribute : Attribute
{
    public int MaxDegreeOfParallelism { get; }

    public SupportsConcurrencyAttribute(int maxDegreeOfParallelism) => MaxDegreeOfParallelism = maxDegreeOfParallelism;

    public SupportsConcurrencyAttribute() => MaxDegreeOfParallelism = Environment.ProcessorCount;
}
```
If a developer needs more control, they can still adjust settings as needed. I think this should be clearly documented also. 

We might consider tweaking the priority scheduling algorithm for jobs triggered to run immediately, treating them as high priority tasks, bumping them up the queue. Additionally, a 'force execution' override could be useful for bypassing concurrency limits and ensuring the job runs immediately. Let me know your thoughts.

It also makes sense to integrate some of these functionalities directly into the JobExecutor, especially those that don't require scheduling but still benefit from retries etc.

Reply:
PS: Saw that you used `* 4` so ignore that comment.